src/content/ui/adaptive-responsive/best-practices.md

```diff
@@ -0,0 +1,232 @@
+Flutter can run on many different platforms,
+and your app might be running in a resizeable window on ChromeOS,
+side by side with another app on tablets in a multi-window mode,
+or even in a picture-in-picture on phones.
+Therefore, device type and app window size aren’t
+really strongly connected.
+
+Instead, use `MediaQuery` to get the size of the window
+your app is currently running in.
+
+This isn’t only helpful for UI code.
+To learn how abstracting out device
+capabilities can help your business logic code,
+check out the 2022 Google I/O talk,
+[Flutter lessons for federated plugin development][].
+ 
+[Flutter lessons for federated plugin development]: {{site.youtube-site}}/watch?v=GAnSNplNpCA
+
+### Support a variety of input devices
+
+Apps should support basic mice, trackpads,
+and keyboard shortcuts. The most common user
+flows should support keyboard navigation
+to ensure accessibility. In particular,
+your app follow accessible best practices
+for keyboards on large devices.
+
+The Material library provides widgets with
+excellent default behavior for touch, mouse,
+and keyboard interaction.
+
+To learn how to add this support to custom widgets,
+check out [User input & accessibility][].
+
+[User input & accessibility]: /ui/adaptive-responsive/input
+
+### Restore List state
+
+<b>PENDING: Reid, you suggested renaming/removing? this and I can't, for the life of me, fine that comment in the PR</b>
+
+To maintain the scroll position in a list
+that doesn't change its layout when the
+device's orientation changes,
+use the [`PageStorageKey`][] class.
+[`PageStorageKey`][] persists the
+widget state in storage after the widget is
+destroyed and restores state when recreated.
+
+You can see an example of this in the [Wonderous app][],
+where it stores the list's state in the
+`SingleChildScrollView` widget.
+
+If the `List` widget changes its layout
+when the device's orientation changes,
+you might have to do a bit of math ([example][])
+to change the scroll position on screen rotation.
+
+[example]: {{site.github}}/gskinnerTeam/flutter-wonderous-app/blob/34e49a08084fbbe69ed67be948ab00ef23819313/lib/ui/screens/collection/widgets/_collection_list.dart#L39
+[`PageStorageKey`]: {{site.api}}/flutter/widgets/PageStorageKey-class.html
+[Wonderous app]: {{site.github}}/gskinnerTeam/flutter-wonderous-app/blob/8a29d6709668980340b1b59c3d3588f123edd4d8/lib/ui/screens/wonder_events/widgets/_events_list.dart#L64
+
+# Save app state
+
+Apps should retain or restore [app state][]
+as the device rotates, changes window size,
+or folds and unfolds. 
+By default, Flutter should maintain state.
+
+If your app loses state during device configuration,
+verify that the plugins and native extensions
+that your app uses support the
+device type, such as a large screen.
+Some native extensions might lose state when the
+device changes position.
+
+<b>TODO(sfshaza): PENDING: What can you do when this happens?</b>
```

Comment:
in Wonderous's case they used a different plugin that didnt have the issue... but if there isn't another plugin option, then filing an issue to that plugin/package.

I believe you got this best practice from my blog post so here it is with more details on what happened. https://medium.com/flutter/developing-flutter-apps-for-large-screens-53b7b0e17f10#:~:text=Problem%3A%20Folding/Unfolding%20causes%20state%2Dloss

Reply:
Super helpful!